### PR TITLE
FFmpeg deprecate old API methods

### PR DESCRIPTION
**Pull request by fritsch**
This is a placeholder, so that no one needs to start from scratch again. DVDVideoCodecFFmpeg and DVDDemuxFFmpeg I left out, cause these need major rework, including the hw decoder to react on EAGAIN.

### CODE REVIEW

**Comment by FernetMenta** [comment]
please bump ffmpeg in a separate pr

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 1]
```diff
@@ -3021,12 +3020,19 @@ IAESound *CActiveAE::MakeSound(const std::string& file)
...
     // decode until eof
     av_init_packet(&avpkt);
-    int len;
+    int ret = 0;
     while (av_read_frame(fmt_ctx, &avpkt) >= 0)
     {
       int got_frame = 0;
-      len = avcodec_decode_audio4(dec_ctx, decoded_frame, &got_frame, &avpkt);
-      if (len < 0)
+      ret = avcodec_send_packet(dec_ctx, &avpkt);
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp]: I don't like this. a return value of a function should be checked for error right after the call, not 10 lines after.

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 2]
```diff
@@ -3021,12 +3020,19 @@ IAESound *CActiveAE::MakeSound(const std::string& file)
...
+
+      if (ret >= 0)
+        ret = avcodec_receive_frame(dec_ctx, decoded_frame);
+
+      if (ret >= 0)
+        got_frame = 1;
+
+      if (ret < 0)
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp]: that is still wrong btw. AVERROR(EGAIN) should not be treated as an error. that hapens becuase you try to reuse the same ret code for 2 different functions. keep it simple

**Comment by fritsch** [comment]
Bump separated: https://github.com/xbmc/xbmc/pull/10039

**Comment by FernetMenta** [comment]
> DVDVideoCodecFFmpeg and DVDDemuxFFmpeg I left out, cause these need major rework, including the hw decoder to react on EAGAIN.

this is btw not correct. only minimal adjustments are required.

**Comment by fritsch** [comment]
Okay - then I am thankful for a PR for these two.

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 3]
```diff
@@ -3059,6 +3049,30 @@ IAESound *CActiveAE::MakeSound(const std::string& file)
         delete sound;
         return NULL;
       }
+
+      if (sent != -1)
+      {
+        receive_frame_error = avcodec_receive_frame(dec_ctx, decoded_frame);
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp]: I think this this method could be simplified a lot if it handled read and send in a loop, then receive in a separate one

**Comment by fritsch** [comment]
Obsolete - WIP

**Pull request closed without merging**